src/Symfony/Bundle/SecurityBundle/Tests/Functional/Bundle/RememberMeBundle/Security/UserChangingUserProvider.php

```diff
@@ -39,7 +40,7 @@ public function refreshUser(UserInterface $user)
     {
         $user = $this->inner->refreshUser($user);
 
-        $alterUser = \Closure::bind(function (User $user) { $user->password = 'foo'; }, null, User::class);
+        $alterUser = \Closure::bind(function (InMemoryUser $user) { $user->password = 'foo'; }, null, class_exists(User::class) ? User::class : InMemoryUser::class);
```

Comment:
sure about the closure typehint? the class `InMemoryUser` always exists?

Reply:
Yes, the class was introduced in 5.3 (which is also the minimum version of in the requirements of this bundle).

This problem is actually a bit complex: The `$password` property is private, so we must bind the closure to the class holding the private property. For BC reasons, this class is `User` in <6.0. For 6.0+, it's `InMemoryUser` (as the old class is removed). So we have to conditionally bind to the correct user class.